### PR TITLE
cast float when denormalizing float16 tensor

### LINKED ISSUE #1615: show_results() not working when called on a fp16 Learner
**Description**
`show_results()` crashes when called on a half-precision  `Learner`.

**Installation details**
```text
=== Software === 
python        : 3.7.1
fastai        : 1.0.42
fastprogress  : 0.1.18
torch         : 1.0.1.post2
nvidia driver : 410.78
torch cuda    : 10.0.130 / is available
torch cudnn   : 7402 / is enabled

=== Hardware === 
nvidia gpus   : 1
torch devices : 1
  - gpu0      : 7929MB | GeForce RTX 2070

=== Environment === 
platform      : Linux-4.15.0-45-generic-x86_64-with-debian-buster-sid
distro        : #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019
conda env     : fastai
python        : /home/weenkus/anaconda3/envs/fastai/bin/python
sys.path      : /home/weenkus/Workspace/Training/course-v3/nbs/dl1
/home/weenkus/anaconda3/envs/fastai/lib/python37.zip
/home/weenkus/anaconda3/envs/fastai/lib/python3.7
/home/weenkus/anaconda3/envs/fastai/lib/python3.7/lib-dynload
/home/weenkus/anaconda3/envs/fastai/lib/python3.7/site-packages
/home/weenkus/anaconda3/envs/fastai/lib/python3.7/site-packages/IPython/extensions
```

**To Reproduce**
I managed to reproduce it by adding these tests to `test_vision_train.py`;

```python
# PASSES
def test_show_results(learn):
    learn.show_results()

# PASSES
def test_show_results_16bit_to_32bit(learn):
    learn.to_fp16()
    learn.to_fp32()
    learn.show_results()

# CRASHES
def test_show_results_32bit(learn):
    learn.to_fp16()
    learn.show_results()
```

The first two tests pass with 32bit models while the third tests crashes because of the half-precision model.

**Expected behavior**
I expected a half-precision learner after training to be able to call the method `show_results()`

**Screenshots**
Reproduced it via the lesson 1 (part 1 2019 course) jupyter notebook.
Screenshot image

Code;
```python
# CELL 1
%reload_ext autoreload
%autoreload 2
%matplotlib inline

from fastai.vision import *
from fastai.metrics import error_rate

path = untar_data(URLs.PETS)

path_anno = path/'annotations'
path_img = path/'images'
fnames = get_image_files(path_img)
pat = r'/([^/]+)_\d+.jpg$'

data = ImageDataBunch.from_name_re(
    path_img, fnames, pat, ds_tfms=get_transforms(), size=224, bs=64
).normalize(imagenet_stats)

# CELL 2
learn = create_cnn(data, models.resnet34, metrics=error_rate)
learn.fit_one_cycle(1)
learn.show_results(rows=1)

# CELL 3
learn = create_cnn(data, models.resnet34, metrics=error_rate)
learn.to_fp16()
learn.fit_one_cycle(1)
learn.show_results(rows=1)
```

### PR DESCRIPTION
**Pull request by sunhwan**
Fixes #1615

### CODE REVIEW

File: fastai/vision/data.py [hunk 1]
```diff
@@ -58,7 +58,10 @@ def normalize(x:TensorImage, mean:FloatTensor,std:FloatTensor)->TensorImage:
 
 def denormalize(x:TensorImage, mean:FloatTensor,std:FloatTensor, do_x:bool=True)->TensorImage:
     "Denormalize `x` with `mean` and `std`."
-    return x.cpu()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
+    if x.dtype == torch.float16:
+        return x.cpu().float()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
+    else:
```
Comment (sgugger) [inline fastai/vision/data.py]: You don't need to do a test: x.cpu().float() won't do anything if x is already in FP32 precision.

**Comment by sgugger** [comment]
There is still a bug with this, you're right. Thanks for your fix! We can do it more simply though, request a change before merging.

File: fastai/vision/data.py (hunk 1 above)
Comment (sunhwan) [reply fastai/vision/data.py]: Got it. I've changed it as suggested.

**Comment by sgugger** [comment]
Perfect, thanks!

**Pull request merged**